client/packages/common/src/ui/layout/tables/components/Header/ColumnPicker.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { useEffect } from 'react';
+import {
+  IconButton,
+  Tooltip,
+  Popover,
+  Stack,
+  Typography,
+  FormControlLabel,
+} from '@mui/material';
+import {
+  Checkbox,
+  Column,
+  ColumnsIcon,
+  RecordWithId,
+  useLocalStorage,
+} from '@openmsupply-client/common';
+import { LocaleKey, useTranslation } from '@common/intl';
+
+interface ColumnPickerProps<T extends RecordWithId> {
+  columnKey: string;
+  columns: Column<T>[];
+  onChange: (columns: Column<T>[]) => void;
+}
+
+export const ColumnPicker = <T extends RecordWithId>({
+  columnKey,
+  columns,
+  onChange,
+}: ColumnPickerProps<T>) => {
+  const t = useTranslation('common');
+  const [hiddenColumns, setHiddenColumns] = useLocalStorage('/columns/hidden');
+  const [anchorEl, setAnchorEl] = React.useState<HTMLButtonElement | null>(
+    null
+  );
+
+  const open = Boolean(anchorEl);
+  const id = open ? 'simple-popover' : undefined;
+
+  const getHiddenColumns = () =>
+    hiddenColumns ? hiddenColumns[columnKey] ?? [] : [];
```

Comment:
hmm, this is a little bit confusing for me, i think maybe nam hiddenColumns to be a bit more specific ? `hiddenColumns` -> `hiddenColumnsConfig` and `columnKey` -> `tableKey` (or something like that).

Also here can we just `hiddenColumnsConfig?.[tableKey] || []`

Reply:
yeah, thanks. it bothered me too - have put in your suggestions.